Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -1566,4 +1566,76 @@ function Set-ToolsRepo
             Write-Host "Successfully set tools-repo on all hosts for datastore $ds_name"
         }
     }
-}
\ No newline at end of file
+}
+
+<#
+.Synopsis
+    Set vSAN compression and deduplication on a cluster or clusters. If deduplication is enabled then compression is required.
+    The default cluster configuration is deduplication and compression but the customer can change that.
+    Choosing neither compression nor deduplication will disable both.
+    This requires action on every physical disk and will take time to complete.
+
+.EXAMPLE
+    Set-vSANCompressDedupe -Cluster "cluster-1,cluster-2" -Compression $true
+
+    Set-vSANCompressDedupe -Cluster "cluster-1,cluster-2" -Deduplication $true
+
+    Set-vSANCompressDedupe -Cluster "cluster-1,cluster-2"
+
+    Set-vSANCompressDedupe -Cluster "all"
+#>
+
+function Set-vSANCompressDedupe
+{
+    [AVSAttribute(120, UpdatesSDDC = $true)]
+    param(
+        [Parameter(Mandatory = $true)]
+        [String]$Cluster,
+        [Parameter(Mandatory = $false,
+        HelpMessage = "Enable compression and deduplication.")]
+        [bool]$Deduplication = $false,
+        [Parameter(Mandatory = $false,
+        HelpMessage = "Enable compression only.")]
+        [bool]$Compression = $true,
+        [Parameter(Mandatory = $false,
+        HelpMessage = "Neither compression nor deduplication.")]
+        [bool]$Neither = $false
+    )
+
+    # $cluster is an array of cluster names or "all"
+    If ($Cluster -eq "all")
+    {
+        $Clusters = Get-Cluster
+    }
+    else
+    {
+        foreach ($cluster_each in ($Cluster.split(",")).Trim())
```

Comment:
There's an option in Split to skip empty entries, probably should use that as well.

Reply:
Added